classes/checkout/PaymentOptionsFinder.php

```diff
@@ -40,7 +40,7 @@ public function find() //getPaymentOptions()
     {
         // Payment options coming from intermediate, deprecated version of the Advanced API
         $this->hookName = 'displayPaymentEU';
-        $rawDisplayPaymentEUOptions = parent::find();
+        $rawDisplayPaymentEUOptions = array_filter(parent::find(), 'is_array');
```

Comment:
I'm not sure it's the good way to fix the bug.
Regarding the issue the problem come from the module, this one shouldn't return a null value for this hook.
It's the module developer responsibility to use the correct behaviour :thinking:

Reply:
But his does not prevent us from checking if it returns the expected value !

Reply:
You can check the return and throw a more adapted exception to help identify the problem, but instead the problem should be ignored/hidden

Reply:
@PierreRambaud This is not a bug fix, but rather an improvement to avoid fatal errors, in case the developer has obulated to return the expected value ... wdyf ?

Reply:
Hi @jolelievre

Reply:
Maybe trigger a php warning?

Reply:
The thing is that the error should not be hidden if there is one, and a warning will be ignored by most merchants/devs as they don't look into the logs
If you ignore the problem, then the only thing that will happen is people won't understand that it doesn't work. Usually when an unexpected behaviour is occurring and there is no possible fallback it's best to have an error, the more detailed the error the easier the debugging can then be done but it's still better to have an error

Reply:
I don't consider this as an improvement, because you add treatments and use memory for something that should never happened. Wasting memory if the first step of bad performance :/

Reply:
> ... for something that should never happened




Are you sure @PierreRambaud  🤔 ? See this case please https://github.com/PrestaShop/PrestaShop/issues/18229

Reply:
This is what I said, if you use this hook, you know you shouldn't return null but an array.